backend/src/routes/api/integrations/nim/index.ts

```diff
@@ -62,35 +61,72 @@ module.exports = async (fastify: KubeFastifyInstance) => {
+          // If the secret already exists, ensure the account is created or retrieved
+          if (!createdNew) {
+            try {
+              const account = await getNIMAccount(fastify);
+              if (!account) {
+                // If the account does not exist, create it
+                const response = await createNIMAccount(fastify);
                 const isEnabled = isAppEnabled(response);
                 reply.send({
                   isInstalled: true,
                   isEnabled: isEnabled,
                   canInstall: false,
                   error: '',
                 });
-              })
-              .catch((e) => {
-                const message = `Failed to create NIM account, ${e.response?.body?.message}`;
-                fastify.log.error(message);
-                reply.status(e.response.statusCode).send(new Error(message));
+              } else {
+                // If the account exists, get its status
+                const isEnabled = isAppEnabled(account);
+                reply.send({
+                  isInstalled: true,
+                  isEnabled: isEnabled,
+                  canInstall: false,
+                  error: '',
+                });
+              }
+            } catch (accountError: any) {
+              const message =
+                accountError.response?.statusCode === 409
+                  ? 'NIM account already exists.'
+                  : `Failed to create or retrieve NIM account: ${accountError.response?.body?.message}`;
+              fastify.log.error(message);
+              reply.status(accountError.response?.statusCode || 500).send(new Error(message));
+            }
+          } else {
+            // If the secret is newly created, ensure the account is created
+            try {
+              const response = await createNIMAccount(fastify);
+              const isEnabled = isAppEnabled(response);
+              reply.send({
+                isInstalled: true,
+                isEnabled: isEnabled,
+                canInstall: false,
+                error: '',
               });
-          })
-          .catch((e) => {
-            if (e.response?.statusCode === 409) {
-              fastify.log.error(`NIM secret already exists, skipping creation.`);
-              reply.status(409).send(new Error(`NIM secret already exists, skipping creation.`));
-            } else {
-              fastify.log.error(`Failed to create NIM secret. ${e.response?.body?.message}`);
-              reply
-                .status(e.response.statusCode)
-                .send(new Error(`Failed to create NIM secret, ${e.response?.body?.message}`));
+            } catch (accountError: any) {
+              const message = `Failed to create NIM account: ${accountError.response?.body?.message}`;
+              fastify.log.error(message);
+              reply.status(accountError.response?.statusCode || 500).send(new Error(message));
             }
-          });
+          }
```

Comment:
Okay cool, now that the `if (!createdNew) ...` supports the creationion of a new account as well, I think we can remove the entire `if ... else ...` and only keep the code block that was inside the if statement.